/*
    Copyright (C) 2020 by the NebulaStream project (https://nebula.stream)

    Licensed under the Apache License, Version 2.0 (the "License");
    you may not use this file except in compliance with the License.
    You may obtain a copy of the License at

        https://www.apache.org/licenses/LICENSE-2.0

    Unless required by applicable law or agreed to in writing, software
    distributed under the License is distributed on an "AS IS" BASIS,
    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
    See the License for the specific language governing permissions and
    limitations under the License.
*/

syntax = "proto3";
package NES;


/*
The serializable wrapper QueryReconfigurationPlan
 */
message SerializableQueryReconfigurationPlan {

  enum QueryReconfigurationType {
    DATA_SOURCE = 0;
    DATA_SINK = 1;
  }

  uint64 id = 1;
  uint64 queryId = 2;
  QueryReconfigurationType queryReconfigurationType = 3;
  uint64 oldQuerySubPlanId = 4;
  uint64 newQuerySubPlanId = 5;
}